### ISSUE TITLE
Searching for replacements of markdown crate

### ISSUE THREAD

**Issue by tesuji** [rust-lang/rustup#1782]
**Describe the problem you are trying to solve**

I'm searching for replacements for [`markdown`] crate.

Rationales: 
- Latest commit of markdown: Oct 27, 2017.
- It almost be abandonned.

**Describe the solution you'd like**

There are two solutions here:

1. Find new crates to replace markdown crate.
   Candidates: 
   * [`pulldown-cmark`]
   * [`comrak`]
1. Fork markdown crate and keep it up-to-date.
  We then specify dependencies like this in `Cargo.toml`:
   ```toml
   markdown = { version = "0.2", git = "https://github.com/rust-lang-nursery/markdown" }
    ```

**Drawbacks**

The first solution seems really difficult: 
`rustup` currently needs to iterate through `markdown::Block`
to print content to `std{out,err}` appropriately. But `pulldown_cmark`
doesn't provide equivalent methods (only `push_html` and `write_html`).

[`markdown`]: https://github.com/johannhof/markdown.rs
[`comrak`]: https://docs.rs/comrak/0.5.1/comrak/
[`pulldown-cmark`]: https://docs.rs/pulldown-cmark/

**Comment by kinnison** [comment]
Do we truly need to update the markdown crate?  We only use it for processing chunks of text for our output routines, and the version we have is sufficient for our needs.

**Comment by tesuji** [comment]
What if markdown crate has security vulnerable but no one care enough
to find out and fix it? At least, we should keep `markdown`s dependencies up to date.

**Comment by kinnison** [comment]
Given that it's pretty much only used to parse content under our control (i.e. text inside the tool) I'm not too worried about dependencies for now.  `pulldown_cmark` I've used in other projects and while it might be a bit more of a pain to work with for us, it's pretty good.

**Comment by tesuji** [comment]
Closed by #2009.
